.travis.yml

```diff
@@ -0,0 +1,3 @@
+language: nix
+script:
+  - nix-shell --command "GOCACHE= go test ./..."
```

Comment:
Why did you add `GOCACHE=`? This may break with Go 1.12: https://golang.org/doc/go1.11#gocache

This does not work with `nix-shell --pure`, so it probably won't work in Travis.

Reply:
Because the nix-shell set `GOCACHE=off`:

```console
$ nix-shell --command "echo $GOCACHE" 
off
```

This would break the build in this case.

Reply:
`nix-shell --pure` does not allow network access, travis does.

Reply:
It now sets `GOCACHE=` to an tempdir instead. Also `GOCACHE=` should still work with go 1.12 because it does not disable the cache dir and rather use an default value.